### ISSUE TITLE
[no-require-imports] `require` created by `createRequire` should be ignored

### ISSUE THREAD

**Issue by JounQin** [typescript-eslint/typescript-eslint#3713]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-require-imports": 2
  }
}
```

```TS
import { createRequire } from 'module'

const require = createRequire()

require('remark-preset-prettier')
```

**Expected Result**

No error

**Actual Result**

```log
error  A `require()` style import is forbidden   @typescript-eslint/no-require-imports
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.29.0` |
| `@typescript-eslint/parser`        | `4.29.0` |
| `TypeScript`                       | `4.3.5` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `12.22.1` |

**Comment by JounQin** [comment]
Or, that rule should only care about global reference of `require`.

What means `const require = (input: string) => {}; require('')` should just be fine?

**Comment by rafaelss95** [comment]
Since this rule simply checks the AST, which means there's **no** type information, there is no way it can know if it's a "global require" or not and so in this case, IMHO the best you can do is disable it with a comment.

**Comment by bradzacher** [comment]
Through the use of scope analysis you can do accurate detection of cases like this!

If a variable has no declarations, then it's a global variable. So when we inspect a `require` call, we can just get the scope analysis variable and for the name in the scope and check if has a declaration. If it does, it's a local variable like in OP. If not then it's a global variable.

**Comment by rafaelss95** [comment]
Oh yeah. Sorry, I completely forgot about `Scope` :man_facepalming:
